Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/FileAccessPolicyTests.cs

```diff
@@ -70,20 +70,12 @@ public void ValidateCachingUntrackedPathInput(bool declareDependency, bool input
 
             // Modify untracked input file /dir/nestedFile
             File.WriteAllText(ArtifactToString(nestedFile), "nestedFile");
-            if (declareDependency)
-            {
-                // Changes in declared dependencies always cause cache misses

```

Comment:
With this PR, the semantics tested here clearly changes.

I don't know how important this test is.  It is a bit scary to see that if an explicitly declared dependency changes the pip is not invalidated.  After all, maybe it's better to exclude only the executable path from the fingerprint when it's untracked, instead of every single dependency.

@narasamdya, what do you think?  I'm leaning towards not breaking this test.

Reply:
I'm leaning towards consistency. I think this PR changes the semantics, where untracked stuff overrides dependencies. 

I don't mind we're breaking this test.

However, I'm still thinking that the changes should go to the process builder.

---
In reply to: [333088339](https://github.com/microsoft/BuildXL/pull/997#discussion_r333088339) [](ancestors = 333088339)

Reply:
Ok, let's just update the doc comment for this test method to reflect the new semantics